code/game/objects/items/weapons/tools/upgrades.dm

```diff
@@ -0,0 +1,445 @@
+	icon_state = "heatsink"
+	prefix = "heatsunk"
+	degradation_mult = 0.65
+
+
+/obj/item/weapon/tool_upgrade/reinforcement/heatsink/can_apply(var/obj/item/weapon/tool/T, var/mob/user)
+	.=..()
+	if (.)
+		if (T.use_fuel_cost || T.use_power_cost)
+			return TRUE
+		return FALSE
+
+/obj/item/weapon/tool_upgrade/reinforcement/plating
+	name = "reinforced plating"
+	desc = "A sturdy bit of metal that can be bolted onto any tool to protect it. Tough, but bulky"
+	icon_state = "plate"
+	prefix = "reinforced"
+	degradation_mult = 0.55
+	force_mod = 1
+	precision = -5
+	bulk_mod = 1
+
+
+/obj/item/weapon/tool_upgrade/reinforcement/guard
+	name = "metal guard"
+	desc = "A bent piece of metal that wraps around sensitive parts of a tool, protecting it from impacts, debris, and stray fingers."
+	icon_state = "guard"
+	required_qualities = list(QUALITY_CUTTING,QUALITY_DRILLING, QUALITY_SAWING, QUALITY_DIGGING, QUALITY_EXCAVATION, QUALITY_WELDING)
+	prefix = "shielded"
+	degradation_mult = 0.75
+	precision = 5
+
+
+// 	 PRODUCTIVITY: INCREASES WORKSPEED
+//------------------------------------------------
+/obj/item/weapon/tool_upgrade/productivity/ergonomic_grip
+	name = "ergonomic grip"
+	desc = "A replacement grip for a tool which allows it to be more precisely controlled with one hand"
+	icon_state = "ergonomic"
+	prefix = "ergonomic"
+	workspeed = 0.15
+
+
+/obj/item/weapon/tool_upgrade/productivity/ratchet
+	name = "ratcheting mechanism"
+	desc = "A mechanical upgrade for wrenches and screwdrivers which allows the tool to only turn in one direction"
+	icon_state = "ratchet"
+	required_qualities = list(QUALITY_BOLT_TURNING,QUALITY_SCREW_DRIVING)
+	prefix = "ratcheting"
+	workspeed = 0.25
+
+/obj/item/weapon/tool_upgrade/productivity/red_paint
+	name = "red paint"
+	desc = "Do red tools really work faster, or is the effect purely psychological"
+	icon_state = "paint_red"
+	prefix = "red"
+	workspeed = 0.20
+	precision = -10
+
+/obj/item/weapon/tool_upgrade/productivity/red_paint/apply_values()
+	if (..())
+		holder.color = "#FF4444"
+
+/obj/item/weapon/tool_upgrade/productivity/whetstone
+	name = "sharpening block"
+	desc = "A rough single-use block to sharpen a blade. The honed edge cuts smoothly"
+	icon_state = "whetstone"
+	required_qualities = list(QUALITY_CUTTING,QUALITY_SAWING, QUALITY_WIRE_CUTTING)
+	prefix = "sharpened"
+	workspeed = 0.15
+	precision = 5
+	force_mult = 1.15
+
+
+/obj/item/weapon/tool_upgrade/productivity/diamond_blade
+	name = "Asters \"Gleaming Edge\": Diamond blade"
+	desc = "An adaptable industrial grade cutting disc, with diamond dust worked into the metal. Exceptionally durable"
+	icon_state = "diamond_blade"
+	required_qualities = list(QUALITY_CUTTING,QUALITY_SAWING, QUALITY_WIRE_CUTTING, QUALITY_PRYING)
+	prefix = "diamond-edged"
+	workspeed = 0.25
+	degradation_mult = 0.85
+	force_mult = 1.10
+	matter = list(MATERIAL_STEEL = 1, MATERIAL_DIAMOND = 1)
+
+
+/obj/item/weapon/tool_upgrade/productivity/oxyjet
```

Comment:
This upgrade needs more degradation multiplier, to make tool with this installed less reliable.